src/components/forms/AddSeparatorForm.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <section class="general-form">
+    <h2>Separadores</h2>
+    <div v-if="updatingShelf">
+      <form @submit.prevent="updateSeparators">
+        <div class="furniture-input">
+          <label>Cuantos Separadores ?</label>
+          <select
+            name="amountOfSeparators"
+            id="amountOfSeparators"
+            v-model.number="amountOfSeparators"
+            @change="updateSeparators"
+          >
+            <option value="0">0</option>
+            <option value="1">1</option>
+            <option value="2">2</option>
+            <option value="3">3</option>
+          </select>
+        </div>
+        <div class="furniture-input">
+          <label>Que orientacion ?</label>
+          <select
+            name="typeOfSeparators"
+            id="typeOfSeparators"
+            v-model="typeOfSeparators"
+            @change="updateSeparators"
+          >
+            <option value="centered">centrado</option>
+            <option value="left">izquierda</option>
+            <option value="right">derecha</option>
+          </select>
+        </div>
+        <button>Confirmar</button>
+      </form>
+    </div>
+    <div v-else>
+      <h2>Haz click en el espacio para agregar separadores</h2>
+    </div>
+  </section>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+export default {
+  props: ['myUpdatingShelf'],
+  data() {
+    return {
+      amountOfSeparators: 0,
+      typeOfSeparators: 'centered',
+    };
+  },
+  computed: {
+    ...mapGetters(['shelfs']),
+    updatingShelf() {
+      return this.shelfs.find((shelf) => shelf === this.myUpdatingShelf);
```

Comment:
Ah I just understood that updating shelf is a computed value and not a function.
So clearly, the name is confusion. 

Just name it `shelf()`

Reply:
actually, I thing you should do something else.

I thing the props should give you the full shelf object. 
Then the component should modify it. 
And at the end (in updateSeparators), you should have a fonction that "saves" the shelf in the store.

It's important in code to avoid "side effects" and "spagetthi code". 

What that means is that we try to keep the interconections and "cables" clean and not have everything interconected in all directions. 

A good way to do it is to keep your components independent. 

Working alone with what it has received as props.

And only marginally, a minimum amount of times, having some "side effects" like "saving data into store".

Think about a chef in a kitchen: he doesn't cook inside the fridge. He takes at the begining what it needs from the fridge. At the end he puts back the stuff in the fridge. But he is not chopping a tomato inside the fridge.

Reply:
the store is your fridge

Reply:
#metaphorsMatter